scripts/run_diff.sh

```diff
@@ -76,17 +76,22 @@ git_start=`git merge-base $git_start $git_merge_base_end`
 
 cleanup()
 {
-  rm -f $diff_file
+  rm -f $diff_file $added_lines_file
 }
 
 trap cleanup EXIT
 
 diff_file=`mktemp`
+added_lines_file=`mktemp`
 
-git diff $git_start $git_end > $diff_file
+# Pass the output through iconv to remove any invalid UTF-8 (diff_to_added_lines.py will die otherwise)
 
-# Get the list of files that have changed
-diff_files=`git diff --name-only $git_start $git_end`
+git diff $git_start $git_end | iconv -t utf-8 -c > $diff_file
+
+# Get the list of files that have changed, that end with lintable extensions
+diff_files=`git diff --name-only $git_start $git_end | grep "\.\(\(cpp\)\|\(hh\)\|\(cc\)\|h\)$"`
+
+$script_folder/diff_to_added_lines.py $diff_file $absolute_repository_root > $added_lines_file
```

Comment:
Consider running `shellcheck` on this entire file. This line could be written less ambiguously:
first item should be `"${script_folder}/diff_to_added_lines.py"` for example.